frontend/src/services/api/__tests__/client.test.ts

```
@@ -0,0 +1,451 @@
+          data: { message: 'Success after refresh' },
+          status: 200,
+        });
+
+        // Import fresh to trigger interceptor setup
+        const { default: client } = await import('../config');
+        Object.assign(mockAxiosInstance, client);
+
+        const result = await responseInterceptor.error(unauthorizedError);
+
+        expect(mockAuthStore.refreshToken).toHaveBeenCalled();
+        expect(unauthorizedError.config!.headers!.Authorization).toBe(`Bearer ${mockNewToken}`);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)